core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -58,10 +76,9 @@ public Schema record(Schema record, List<String> names, List<Schema> fields) {
       // case where the converted field is non-null is when a map or list is
       // selected by lower IDs.
       if (selectedIds.contains(fieldId)) {
-        filteredFields.add(copyField(field, field.schema()));
+        filteredFields.add(copyField(field, field.schema(), fieldId));
       } else if (fieldSchema != null) {
-        hasChange = true;
-        filteredFields.add(copyField(field, fieldSchema));
+        filteredFields.add(copyField(field, fieldSchema, fieldId));
```

Comment:
This should still set `hasChange = true`, right?

Reply:
You are right!